common/reflect_resource.go

```diff
@@ -617,10 +621,16 @@ func collectionToMaps(v any, s *schema.Schema, aliases map[string]map[string]str
 			}
 			v = v.Elem()
 		}
-		err := iterFields(v, []string{}, r.Schema, aliases, func(fieldSchema *schema.Schema,
-			path []string, valueField *reflect.Value) error {
+		fv := v.FieldByName("ForceSendFields")
+		forceSendFields := []string{}
+		// force send fields might not exist in struct, so checking it against zero value first
+		if fv != (reflect.Value{}) {
```

Comment:
This is probably the more canonical way to check this.
```suggestion
		if fv.IsValid() {
```

Reply:
> This is probably the more canonical way to check this.

These are not equivalent checks, we need to check that fv is "zero" value (empty reflect.Value struct) which is valid value.